### PR TITLE
test: poetry caching

### PR DESCRIPTION
**Pull request by MartinBernstorff**

### CODE REVIEW

**Comment by MartinBernstorff** [comment]
Environment setup from 2:10 to 10s with a hot cache!

Pretty happy about that. Now our tests take up 85% of the action runtime.

File: .github/workflows/cache_version [hunk 1]
```diff
@@ -0,0 +1,2 @@
+# This file allows manual cache invalidation.
```
Comment (KennethEnevoldsen) [inline .github/workflows/cache_version:1]: Why do we want to specify the version here?

File: .github/workflows/pytest.yml [hunk 2]
```diff
@@ -7,44 +7,57 @@ on:
   pull_request:
     branches: [ main, dev ]
 
+concurrency:
+  # New commit on branch cancels running workflows of the same branch
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
```
Comment (KennethEnevoldsen) [inline .github/workflows/pytest.yml:13]: Love this!

**Review by KennethEnevoldsen — APPROVED** [review]
Looks good.

**Comment by MartinBernstorff** [comment]
How about now @KennethEnevoldsen ?

**Comment by MartinBernstorff** [comment]
@KennethEnevoldsen ...

**Pull request merged**